Automattic/wp-calypso
# CPT: create Add Term component

This branch introduces the `<AddTerm/>` component which allows for creation of new terms for a given taxonomy:

__Default Display__
![new_project_ _testingtimmy2_wordpress_com_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/16813265/a504935c-48e5-11e6-9d92-cf3ca6766126.png)

__Dialog__
![new_project_ _testingtimmy2_wordpress_com_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/16813250/900babd4-48e5-11e6-9c32-4cf4542f97c0.png)

Currently you can add a new term using the dialog, much like the existing "Add Category" feature in the editor sidebar.

__To Test__
- Open up a portfolio project in the editor
- Expand the "Project Types" accordion
- Click the "Add New Project Type" button
- Interact with the modal form.  Use the search box to find a parent.
- Select a parent and note the "Top Level" checkbox unticks
- Play around with validation.  You should not be able to submit a term name that already exists for a given tree level ( i.e. only one top level term could be named `Major Key` ).
- Save a new term, note that it should appear in the original post-editor term selector.  If there are multiple pages of terms, you might need to scroll to trigger a new fetch, or simply search for your newly added term

__Todo in Future PR__
- Auto "checking" newly added term in post that is being edited


Test live: https://calypso.live/?branch=add/term-tree-selector/add-term
